### PR TITLE
add i18n

### PR DESCRIPTION
**Pull request by markkaylor**

### CODE REVIEW

File: template/data/fr/pages.json [hunk 1]
```diff
@@ -0,0 +1,308 @@
+{
+  "pagesFR": [
+    {
+      "id": 5,
+      "slug": "french",
```
Comment (remidej) [inline template/data/fr/pages.json]: I think this should be an empty string, otherwise it won't be the root of the /fr website

But I can see that it's breaking the unique constraint in Strapi. So I really don't don't know what the fix would be right now. I'll come back to it tomorrow

Comment (markkaylor) [reply template/data/fr/pages.json]: Hey thanks for taking a look.  Yeah I agree will need to find workaround for that.  

I should have mentioned the /fr seed data is just a lot of nonsense to check if everything is working.

Comment (remidej) [reply template/data/fr/pages.json]: Yes I figured, nice song btw

File: template/data/fr/global.json [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    "metaTitleSuffix": "Strapi Corporate",
+    "metadata": {
+      "metaTitle": "Strapi starter for Corporate Sites",
+      "metaDescription": "Créer votre site corporate avec Strapi",
+      "twitterCardType": "summary",
+      "twitterUsername": "strapijs",
+      "shareImage": null
+    },
+    "notificationBanner": {
+      "text": "Cette page a été crée avec le Strapi starter pour les sites Corporate.\n [Voir les autres Strapi starters](https://strapi.io/starters)",
```
Comment (remidej) [inline template/data/fr/global.json]: I think it sounds more natural like this in french:

```suggestion
      "text": "Cette page a été crée avec le Strapi starter pour les sites Corporate.\n [Voir les autres starters Strapi](https://strapi.io/starters)",
```

File: template/data/fr/global.json [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            },
+            {
+              "url": "#",
+              "newTab": false,
+              "text": "S'Inscrire"
+            }
+          ]
+        },
+        {
+          "title": "Legal",
```
Comment (remidej) [inline template/data/fr/global.json]:
```suggestion
          "title": "Légal",
```

File: template/data/fr/global.json [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+          "links": [
+            {
+              "url": "#",
+              "newTab": false,
+              "text": "Features"
+            },
+            {
+              "url": "#",
+              "newTab": false,
+              "text": "S'Inscrire"
```
Comment (remidej) [inline template/data/fr/global.json]:
```suggestion
              "text": "S'inscrire"
```

File: template/data/fr/global.json [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        {
+          "url": "/nous-contacter",
+          "newTab": false,
+          "text": "Nous Contacter"
+        }
+      ],
+      "button": {
+        "url": "#",
+        "newTab": false,
+        "text": "S'Inscrire",
```
Comment (remidej) [inline template/data/fr/global.json]:
```suggestion
        "text": "S'inscrire",
```

File: template/data/fr/global.json [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+      "links": [
+        {
+          "url": "/tarifs",
+          "newTab": false,
+          "text": "Tarifs"
+        },
+        {
+          "url": "/nous-contacter",
+          "newTab": false,
+          "text": "Nous Contacter"
```
Comment (remidej) [inline template/data/fr/global.json]: I think if we want to capitalize all links, it should rather be a CSS rule on the frontend

File: template/data/fr/global.json [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+        "text": "S'Inscrire",
+        "type": "secondary"
+      },
+      "logo" : {
+        "image": null,
+        "url": "fr"
+      }
+    },
+    "footer": {
+      "smallText": "© Copyright My Company™",
```
Comment (remidej) [inline template/data/fr/global.json]:
```suggestion
      "smallText": "© Copyright Mon Entreprise™",
```

File: template/data/fr/global.json [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+      "columns": [
+        {
+          "title": "Produit",
+          "links": [
+            {
+              "url": "#",
+              "newTab": false,
+              "text": "Features"
```
Comment (remidej) [inline template/data/fr/global.json]:
```suggestion
              "text": "Fonctionnalités"
```

File: template/data/fr/global.json [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+            },
+            {
+              "url": "#",
+              "newTab": false,
+              "text": "Mentions légales"
+            }
+          ]
+        },
+        {
+          "title": "Company",
```
Comment (remidej) [inline template/data/fr/global.json]:
```suggestion
          "title": "Entreprise",
```

(sounds a bit weird but I couldn't find a better alternative)

File: template/data/fr/global.json [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+            },
+            {
+              "url": "#",
+              "newTab": false,
+              "text": "L'Equipe"
+            }
+          ]
+        },
+        {
+          "title": "Réseau Sociale",
```
Comment (remidej) [inline template/data/fr/global.json]:
```suggestion
          "title": "Réseaux sociaux",
```

File: template/data/fr/global.json [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+          "links": [
+            {
+              "url": "#",
+              "newTab": false,
+              "text": "Nous rejoindre"
+            },
+            {
+              "url": "#",
+              "newTab": false,
+              "text": "L'Equipe"
```
Comment (remidej) [inline template/data/fr/global.json]: I think it should either be `L'Équipe` or `L'équipe`

File: template/data/fr/pages.json [hunk 12]
```diff
@@ -0,0 +1,308 @@
...
+        "metaDescription": "Construire un site modifiable avec Strapi",
+        "twitterCardType": "summary",
+        "twitterUsername": null,
+        "shareImage": null
+      },
+      "contentSections": [
+        {
+          "__component": "sections.hero",
+          "title": "Le meilleure façon de faire votre site Corporate",
+          "label": "Nouveau Strapi starter",
```
Comment (remidej) [inline template/data/fr/pages.json]:
```suggestion
          "label": "Nouveau starter Strapi",
```

File: template/data/fr/pages.json [hunk 13]
```diff
@@ -0,0 +1,308 @@
...
+              "type": "secondary"
+            }
+          ],
+          "picture": null
+        },
+        {
+          "__component": "sections.feature-rows-group",
+          "features": [
+            {
+              "title": "Les équipes marketing peuvent maintenant faire leurs propre pages",
```
Comment (remidej) [inline template/data/fr/pages.json]:
```suggestion
              "title": "Les équipes marketing peuvent maintenant faire leurs propres pages",
```

File: template/data/fr/pages.json [hunk 14]
```diff
@@ -0,0 +1,308 @@
...
+              "description": "Grace à une liste pré-construite de blocs UI, les marketers peuvent faire la mise en page exactement comme ils/elles veulent",
```
Comment (remidej) [inline template/data/fr/pages.json]: Here's one way we can rephrase this to avoid the marketer anglicism, and avoid specifying the gender

```suggestion
              "description": "Grace à une liste pré-construite de blocs UI, les équipes marketing peuvent faire la mise en page exactement comme elles veulent",
```

Marketing has become widely used, but marketer still sounds awkward in french

File: template/data/fr/pages.json [hunk 15]
```diff
@@ -0,0 +1,308 @@
...
+              "link": {
+                "url": "#",
+                "newTab": false,
+                "text": "Voir le liste de blocs UI"
```
Comment (remidej) [inline template/data/fr/pages.json]:
```suggestion
                "text": "Voir la liste de blocs UI"
```

File: template/data/fr/pages.json [hunk 16]
```diff
@@ -0,0 +1,308 @@
...
+              },
+              "media": null
+            },
+            {
+              "title": "Laisser vos développeurs tranquille",
```
Comment (remidej) [inline template/data/fr/pages.json]:
```suggestion
              "title": "Laissez vos développeurs tranquilles",
```

File: template/data/fr/pages.json [hunk 17]
```diff
@@ -0,0 +1,308 @@
...
+              "description": "Vous pouvez publier, modifier, et supprimer des pages sans l'aide d'un développeur.  Votre équipe technique peut finalement concentrer sur d'autre taches.",
```
Comment (remidej) [inline template/data/fr/pages.json]:
```suggestion
              "description": "Vous pouvez publier, modifier, et supprimer des pages sans l'aide d'un développeur.  Votre équipe technique peut enfin se concentrer sur d'autre tâches.",
```

File: template/data/fr/pages.json [hunk 18]
```diff
@@ -0,0 +1,308 @@
...
+        {
+          "__component": "sections.feature-columns-group",
+          "features": [
+            {
+              "title": "Visualisez vos changments",
+              "description": "Grace à un mode preview, vous pouvez voir vos pages avant de les publier.",
+              "icon": null
+            },
+            {
+              "title": "Fully responsive",
```
Comment (remidej) [inline template/data/fr/pages.json]:
```suggestion
              "title": "100% responsive",
```

File: template/data/fr/pages.json [hunk 19]
```diff
@@ -0,0 +1,308 @@
...
+              "description": "Ce starter marche sur touts les tailles d'écrans.",
```
Comment (remidej) [inline template/data/fr/pages.json]:
```suggestion
              "description": "Ce starter marche sur toutes les tailles d'écrans.",
```

File: template/data/fr/pages.json [hunk 20]
```diff
@@ -0,0 +1,308 @@
...
+              "icon": null
+            },
+            {
+              "title": "Facile à personnaliser",
+              "description": "On utilise Tailwind pour le styling. Vous pouvez changez la thème de votre site sans besoin d'aller dans le code.",
```
Comment (remidej) [inline template/data/fr/pages.json]:
```suggestion
              "description": "On utilise Tailwind pour les styles. Vous pouvez changez le thème de votre site sans avoir besoin d'aller dans le code.",
```

File: template/data/fr/pages.json [hunk 21]
```diff
@@ -0,0 +1,308 @@
...
+              "icon": null
+            }
+          ]
+        },
+        {
+          "__component": "sections.testimonials-group",
+          "title": "Ici, les témoignages des vos clients.",
+          "description": "Cette partie est où vous pouvez montrer vos clients.  Mettez des citations, et montrez les logos des entreprises qui aiment votre produit.",
```
Comment (remidej) [inline template/data/fr/pages.json]: Rephrase because there's a word missing, it should be: `Cette partie est là/l'endroit où vous pouvez`

But it's lighter like this:

```suggestion
          "description": "Dans cette partie, vous pouvez mettre en avant vos clients.  Mettez des citations, et montrez les logos des entreprises qui aiment votre produit.",
```

(also avoids using montrer twice)

File: template/data/fr/pages.json [hunk 22]
```diff
@@ -0,0 +1,308 @@
...
+            }
+          ]
+        },
+        {
+          "__component": "sections.lead-form",
+          "title": "S'inscrire à notre Newsletter",
+          "emailPlaceholder": "email@company.com",
+          "submitButton": {
+            "__component": "links.button",
+            "text": "S'Inscrire",
```
Comment (remidej) [inline template/data/fr/pages.json]:
```suggestion
            "text": "S'inscrire",
```

File: template/data/fr/pages.json [hunk 23]
```diff
@@ -0,0 +1,308 @@
...
+        {
+          "id": 2,
+          "locale": "en"
+        }
+      ],
+      "shortName": "Pricing",
+      "status": "published",
+      "metadata": {
+        "metaTitle": "Pricing",
+        "metaDescription": "Les plans différents",
```
Comment (remidej) [inline template/data/fr/pages.json]:
```suggestion
        "metaDescription": "Les différentes offres",
```

File: template/data/fr/pages.json [hunk 24]
```diff
@@ -0,0 +1,308 @@
...
+        "twitterCardType": "summary",
+        "twitterUsername": null,
+        "shareImage": null
+      },
+      "contentSections": [
+        {
+          "__component": "sections.pricing",
+          "title": "Tableaux de tarifs",
```
Comment (remidej) [inline template/data/fr/pages.json]:
```suggestion
          "title": "Tableau de tarifs",
```

File: template/data/fr/pages.json [hunk 25]
```diff
@@ -0,0 +1,308 @@
...
+          "plans": [
+            {
+              "name": "Hobby",
+              "description": "Parfait pour des projets personnel",
+              "isRecommended": null,
+              "price": 0,
+              "pricePeriod": "forever",
```
Comment (remidej) [inline template/data/fr/pages.json]: Should these be localized?

File: template/data/fr/pages.json [hunk 26]
```diff
@@ -0,0 +1,308 @@
...
+        "shareImage": null
+      },
+      "contentSections": [
+        {
+          "__component": "sections.pricing",
+          "title": "Tableaux de tarifs",
+          "plans": [
+            {
+              "name": "Hobby",
+              "description": "Parfait pour des projets personnel",
```
Comment (remidej) [inline template/data/fr/pages.json]:
```suggestion
              "description": "Parfait pour des projets personnels",
```

File: template/data/fr/pages.json [hunk 27]
```diff
@@ -0,0 +1,308 @@
...
+              "features": [
+                {
+                  "name": "Feature 1"
+                },
+                {
+                  "name": "Feature 2"
+                },
+                {
+                  "name": "Feature 3"
+                }
```
Comment (remidej) [inline template/data/fr/pages.json]: Let's keep it in french even if it's not real content

```suggestion
                {
                  "name": "Fonctionnalité 1"
                },
                {
                  "name": "Fonctionnalité 2"
                },
                {
                  "name": "Fonctionnalité 3"
                }
```

File: template/data/fr/pages.json [hunk 28]
```diff
@@ -0,0 +1,308 @@
...
+              "features": [
+                {
+                  "name": "Amazing feature"
+                },
+                {
+                  "name": "Wow effect feature"
+                },
+                {
+                  "name": "Mesmerizing feature"
+                }
```
Comment (remidej) [inline template/data/fr/pages.json]: Let's keep it in french even if it's not real content

```suggestion
                {
                  "name": "Fonctionnalité 1"
                },
                {
                  "name": "Fonctionnalité 2"
                },
                {
                  "name": "Fonctionnalité 3"
                }
```

File: template/data/fr/pages.json [hunk 29]
```diff
@@ -0,0 +1,308 @@
...
+        "metaTitle": "Secret page",
+        "metaDescription": "Preview-only page",
+        "twitterCardType": "summary",
+        "twitterUsername": null,
+        "shareImage": null
+      },
+      "contentSections": [
+        {
+          "__component": "sections.rich-text",
+          "content": "## Page secret. \n\n Vous pouve seulement voir cette page en Preview Mode"
```
Comment (remidej) [inline template/data/fr/pages.json]:
```suggestion
          "content": "## Page secrète. \n\n Vous seul pouvez voir cette page en Preview Mode"
```

File: template/data/fr/pages.json [hunk 30]
```diff
@@ -0,0 +1,308 @@
...
+      "localizations": [
+        {
+          "id": 3,
+          "locale": "en"
+        }
+      ],
+      "shortName": "Secret",
+      "status": "draft",
+      "metadata": {
+        "metaTitle": "Secret page",
```
Comment (remidej) [inline template/data/fr/pages.json]:
```suggestion
        "metaTitle": "Page secrète",
```

File: template/data/fr/pages.json [hunk 31]
```diff
@@ -0,0 +1,308 @@
...
+        "metaTitle": "Nous Contacter",
+        "metaDescription": "Contactez notre équipe",
+        "twitterCardType": "summary",
+        "twitterUsername": null,
+        "shareImage": null
+      },
+      "contentSections": [
+        {
+          "__component": "sections.rich-text",
+          "content": "# Get in touch\n\n> This is an example of a page that relies almost entirely on the RichText section. It's useful for blog articles, or content-heavy pages like legal terms.\n\nWe'd love to hear from you.\n\n## Social media\n\n* [Twitter](#)\n* [Twitter](#)\n* [Twitter](#)\n\n## Postal address\n\n404 Headless Street\n__92210__ **Saint Cloud**, *France*\n\nLorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
```
Comment (remidej) [inline template/data/fr/pages.json]:
```suggestion
          "content": "# Contactez nous\n\n> Ceci est un exemple d'une page qui repose presque entièrement sur le composant RichText. Vous pouvez l'utiliser pour des articles de blog, ou pour des pages comportant beaucoup de texte, comme par exemple les conditions d'utilisation.\n\nNous aimerions beaucoup avoir vos retours.\n\n## Social media\n\n* [Twitter](#)\n* [Twitter](#)\n* [Twitter](#)\n\n## Postal address\n\n404 Headless Street\n__92210__ **Saint Cloud**, *France*\n\nLorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
```

File: template/data/fr/pages.json [hunk 32]
```diff
@@ -0,0 +1,308 @@
...
+        },
+        {
+          "__component": "sections.bottom-actions",
+          "title": "Get in touch",
+          "buttons": [
+            {
+              "url": "#",
+              "newTab": false,
+              "text": "Enovyer un mail",
```
Comment (remidej) [inline template/data/fr/pages.json]:
```suggestion
              "text": "Envoyer un mail",
```

**Review by remidej — CHANGES_REQUESTED** [review]
I hadn't realized there was so much content to translate. Thank you for doing it!

**Comment by remidej** [comment]
@markkaylor thanks for the update, but it looks like there are still quite a few things left from my original review. Make sure you click on "load more" to see everything

**Comment by markkaylor** [comment]
yeah you are right, sorry about that.  Will update first thing in the morning.

File: template/api/page/models/page.settings.json [hunk 33]
```diff
@@ -72,6 +67,14 @@
           "localized": true
         }
       }
+    },
+    "slug": {
+      "pluginOptions": {
+        "i18n": {
+          "localized": true
+        }
+      },
```
Comment (remidej) [inline template/api/page/models/page.settings.json]: I think we should add a regex for validation, because strings are just too easy to break, with characters like spaces for example

Also, my understanding of our conversation with Alex was that we'd stop localizing the slug entirely, to only rely on the locale prefix instead

File: template/data/fr/pages.json (hunk 25 above)
Comment (markkaylor) [reply template/data/fr/pages.json]: yeah I think so

**Review by remidej — APPROVED** [review]
Let's do this 💪
